lib/ansible/module_utils/network/nxos/nxos.py

```diff
@@ -621,6 +622,19 @@ def load_config(self, commands, return_error=False, opts=None, replace=None):
     def edit_config(self, candidate=None, commit=True, replace=None, comment=None):
         resp = list()
 
+        if self._module._name == 'nxos_gir':
+            connection = self._connection
+            if connection.transport == 'local':
+                persistent_command_timeout = C.PERSISTENT_COMMAND_TIMEOUT
```

Comment:
This should also check for `provider['timeout']` value. Something like `persistent_command_timeout = C.PERSISTENT_COMMAND_TIMEOUT or self._module.get(‘provider’, {}).get(‘timeout’)`.

Reply:
Good catch. I'll add a check.